File: benchmarks/cpp/matrix_mult.cpp
// BENCH_MATRIX_SIZE=128
#include <iostream>

static int mat_value(int row, int col, int n) {
    (void)n;
    return (row * 131 + col * 17) % 997;
}

static long long matrix_multiply(int n) {
    long long checksum = 0;
    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < n; ++j) {
            long long total = 0;
            for (int k = 0; k < n; ++k) {
                total += static_cast<long long>(mat_value(i, k, n)) * mat_value(k, j, n);
            }
            checksum += total;
        }
    }
    return checksum;
}

int main() {
    std::cout << matrix_multiply(128) << '\n';
    return 0;
}

File: benchmarks/python/matrix_mult.py
#!/usr/bin/env python3
"""BENCH_MATRIX_SIZE=128"""


def mat_value(row: int, col: int, n: int) -> int:
    return (row * 131 + col * 17) % 997


def matrix_multiply(n: int) -> int:
    checksum = 0
    for i in range(n):
        for j in range(n):
            total = 0
            for k in range(n):
                total += mat_value(i, k, n) * mat_value(k, j, n)
            checksum += total
    return checksum


def main() -> None:
    print(matrix_multiply(128))


if __name__ == "__main__":
    main()
